File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.java
class Solution {
    public long solution(int r1, int r2) {
        long answer = 0;
        
        long count1 = 0;
        for (int y = 1; y < r1; y++) {
            double temp = Math.sqrt((double)r1 * r1 - (double)y * y);
            count1 += (int) temp;
            
            if (temp % 1 == 0)
                count1--;
        }
        
        long count2 = 0;
        for (int y = 1; y < r2; y++) {
            int count = (int) Math.sqrt((double)r2 * r2 - (double)y * y);
            count2 += count;
        }
        answer += (count2 - count1) * 4;
        answer += (r2 - r1 + 1) * 4;
        return answer;
    }
}

File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>

using namespace std;

long long solution(int r1, int r2) {
    long long answer = 0;

    for (long long x = 0; x <= r2; ++x)
    {
        long long inyy = (long long)r1 * r1 - x * x;
        long long outyy = (long long)r2 * r2 - x * x;

        double iny = sqrt(inyy);
        if (x > r1)
            iny = 0;


        double outy = sqrt(outyy);

        iny = ceil(iny);
        outy = floor(outy);

        if (x == 0)
        {
            answer += (outy - iny + 1) * 2;
        }
        else if (iny == 0)
        {
            answer += (outy) * 4;
            answer += 2;
        }
        else
        {
            answer += (outy - iny + 1) * 4;
        }
    }
    return answer;
}